api/v2/checluster_types.go

```diff
@@ -541,6 +541,9 @@ type Gateway struct {
 	// Configuration for kube-rbac-proxy within the Che gateway pod.
 	// +optional
 	KubeRbacProxy *KubeRbacProxy `json:"kubeRbacProxy,omitempty"`
+	// How long the _oauth2_proxy cookie lasts.
+	// +optional
+	CookieExpire string `json:"cookieExpire,omitempty"`
 }
```

Comment:
Thank you for your contirubution.
I believe it is better to add a dedicated structure for additional `oauth-proxy` container configuration:

```go
type Gateway struct {
...
  // +optional
  oAuthProxy *OAuthProxy `json:"oAuthProxy,omitempty"`
}
```

```go
type OAuthProxy struct {
	// Expire timeframe for cookie. If set to 0, cookie becomes a session-cookie which will expire when the browser is closed.
	// +optional
	// +kubebuilder:default:=86400
	// +kubebuilder:validation:Minimum:=0
	CookieExpireSeconds *int32 `json:"cookieExpireSeconds,omitempty"`
}
```